### PR TITLE
merge mkl:bn,lrn,elemwise from dev to mklengine branch unit test pass

### PR DESCRIPTION
**Pull request by rongzha1**
merge mkl_bn, mkl_lrn, mkl_elemwise form dev branch to mklengine branch.
python tests/test_bn, test_elemwise, test_lrn PASS

### CODE REVIEW

File: theano/contrib/mkl/basic_ops.py [hunk 1]
```diff
@@ -737,3 +736,333 @@ def c_code(self, node, name, inp, out, sub):
...
+                }
+
+                status = MKLNdarray_create_buffer_from_primitive(%(z)s, &primitive, dnnResourceSrc);
+                if (status != 0) {
+                    %(fail)s;
+                }
+            }
+
+            if (!dnnLayoutCompare_%(precision)s(layout_user, MKLNdarray_LAYOUT(%(z)s))) {
+                if (NULL == to_internal) {
```
Comment (pengzhao-intel) [inline theano/contrib/mkl/basic_ops.py]: move to_internal allocation into the block of first_run

**Review by pengzhao-intel — COMMENTED** [review]
Move all "to_internal" to first_run

File: theano/contrib/mkl/basic_ops.py (hunk 1 above)
Comment (rongzha1) [reply theano/contrib/mkl/basic_ops.py]: maybe add if (1 == first_run)  like class U2IPool  did in line 610?  
to_internal allocation need to do compare with output %(z)s, but  %(z)s is not in the block of first_run.

**Comment by pengzhao-intel** [comment]
Merge the code and thanks for the works.

**Pull request closed without merging**